snakemake/shell.py

```diff
@@ -132,10 +142,10 @@ def __new__(
             cmd,
             bufsize=-1,
             shell=True,
-            stdout=stdout,
             universal_newlines=iterable or None,
             close_fds=close_fds,
-            **cls._process_args
+            **cls.__get_output_streams(context.get("log"), iterable or read),
```

Comment:
Very nice. What happens if a user uses e.g. stderr="foo.log" and in addition also manually pipes into `{log.stderr}`. I guess since nothing gets captured from here and the manual pipe gets precedence, we are fine, right?

Reply:
That's an extremely good question.  I'll make a test case for this.